### PR TITLE
Support int argument for Dirichlet shape

### PR DESCRIPTION
**Pull request by aryamccarthy**
Currently, cupy's Dirichlet distribution does not support an integer argument a a shape. Only tuples are allowed; ints lead to a `TypeError`. This differs from the behavior of numpy and is not a necessary restriction. It also does not match the [documented parameters](https://docs.cupy.dev/en/stable/reference/generated/cupy.random.dirichlet.html).

The issue has been [detailed on Stack Overflow recently](https://stackoverflow.com/a/66405505/7802200). A workaround is for the user to provide the size as a tuple, but they shouldn't have to.

I address this by turning the provided argument into a list. (If it's already a list or tuple, this won't affect anything.) I provide a test case as well.

Note that cupy's multivariate normal distribution supports both integer and tuple arguments.

### CODE REVIEW

**Review by DimitrisMel — DISMISSED** [review]

File: cupy/random/_generator.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ def dirichlet(self, alpha, size=None, dtype=float):
         if size is None:
             size = alpha.shape
         else:
+            size = list(size)  # In case size is an int
```
Comment (toslunar) [inline cupy/random/_generator.py:146]: `list(5)` causes `TypeError: 'int' object is not iterable`.  How about aligning with this?

https://github.com/cupy/cupy/blob/79085bcdb5ebfecf906a8bc4453dfa0f85bbddb4/cupy/random/_generator.py#L660-L661

```suggestion
            if isinstance(size, int):
                size = size,
```

**Review by toslunar — CHANGES_REQUESTED** [review]

File: cupy/random/_generator.py (hunk 1 above)
Comment (aryamccarthy) [reply cupy/random/_generator.py]: Can do even better by matching multivariate_normal, supporting both `int` and `cupy.integer`. Just pushed that fix.

Why isn't my test case being run, to catch this error?

File: cupy/random/_generator.py [hunk 2]
```diff
@@ -142,6 +142,8 @@ def dirichlet(self, alpha, size=None, dtype=float):
         alpha = cupy.asarray(alpha)
         if size is None:
             size = alpha.shape
+        elif isinstance(size, (int, cupy.integer)):
+            size = [size]
```
Comment (toslunar) [inline cupy/random/_generator.py:146]:
```suggestion
            size = [size] + alpha.shape
```

**Review by toslunar — APPROVED** [review]

**Comment by toslunar** [comment]
pfnCI, test this please.

File: cupy/random/_generator.py (hunk 1 above)
Comment (leofang) [reply cupy/random/_generator.py]: > Why isn't my test case being run, to catch this error?

It's best to test locally. Tests are not run until a reviewer launches CuPy's CI. The GH action etc just test for building and formatting.

File: cupy/random/_generator.py [hunk 3]
```diff
@@ -142,6 +142,8 @@ def dirichlet(self, alpha, size=None, dtype=float):
         alpha = cupy.asarray(alpha)
         if size is None:
             size = alpha.shape
+        elif isinstance(size, (int, cupy.integer)):
+            size = [size] + alpha.shape
```
Comment (toslunar) [inline cupy/random/_generator.py:146]:
```suggestion
            size = (size,) + alpha.shape
```

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by toslunar** [comment]
pfnCI, test this please.

**Review by toslunar — APPROVED** [review]

**Comment by toslunar** [comment]
Thanks, @aryamccarthy, @DimitrisMel.

**Pull request merged**